crates/red_knot/src/symbols.rs

```diff
@@ -274,6 +277,20 @@ impl SymbolTable {
         &self.dependencies
     }
 
+    pub(crate) fn reachable_definitions(
```

Comment:
Can we add some documentation to what this iterator returns. What I understand is that it finds the definitions that are reachable from the current node by navigating backwards through the CFG? How does this work for definitions that aren't defined in the same function?

Reply:
+1 for some more docs here

Reply:
Sure, I can add a docstring here.

> What I understand is that it finds the definitions that are reachable from the current node by navigating backwards through the CFG?

Yes, that's correct; this is the same approach pyright takes.

> How does this work for definitions that aren't defined in the same function?

We only consider local definitions when deciding the type of a symbol. This is necessary for efficient type inference, otherwise the entire program would have to be analyzed i order to decide the type of any module-level symbol.

Assignments to a name from outside its scope should be compatible with its public type. Those assignments will be analyzed when their scope is analyzed.